convert2rhel/unit_tests/utils_test.py

```diff
@@ -347,3 +348,79 @@ def test_prompt_user(question, is_password, response, monkeypatch):
+            {"password": "correct_password", "activation_key": None},
+        ),
+        (
+            "[subscription_manager]\nactivation_key = correct_key\nPassword = correct_password",
+            {"password": "correct_password", "activation_key": "correct_key"},
+        ),
+        (
+            "[subscription_manager]\nincorrect_option = incorrect_content",
+            {"password": None, "activation_key": None},
+        ),
+        ("[INVALID_HEADER]\nactivation_key = correct_key", {"password": None, "activation_key": None}),
+        ("nonexisting", {"password": None, "activation_key": None}),
+    ),
+)
+def test_options_from_config_files_default(monkeypatch, content, output, tmp_path, caplog):
+    """Test config files in default path."""
+    path = os.path.join(str(tmp_path), "convert2rhel.ini")
+    tool_opts = ToolOpts()
+    if content != "nonexisting":
+        with open(path, "w") as file:  # pylint: disable=unspecified-encoding
+            file.write(content)
+        os.chmod(path, 0o600)
+
+    paths = ["/nonexisting/path", path]
+    monkeypatch.setattr(utils, "CONFIG_PATHS", value=paths)
+    tool_opts = utils.options_from_config_files(tool_opts)
+
+    assert tool_opts.password == output["password"]
+    assert tool_opts.activation_key == output["activation_key"]
+    if "INVALID_HEADER" in content:
+        assert "Unsupported header" in caplog.text
+    if "incorrect_option" in content:
+        assert "Unsupported option" in caplog.text
+
+
+@pytest.mark.parametrize(
+    ("content", "output"),
+    (
+        (
+            "[subscription_manager]\nactivation_key = correct_key\nPassword = correct_password",
+            {"password": "correct_password", "activation_key": "correct_key"},
+        ),
+    ),
+)
+def test_options_from_config_files_specified(monkeypatch, content, output, tmp_path, caplog):
+    """Test user specified path for config file."""
+    path = os.path.join(str(tmp_path), "convert2rhel.ini")
+    tool_opts = ToolOpts()
+    with open(path, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content)
+    os.chmod(path, 0o600)
+
+    path_lower_priority = os.path.join(str(tmp_path), "convert2rhel_lower.ini")
+    content_lower_priority = "[subscription_manager]\npassword = low_prior_pass"
+    with open(path_lower_priority, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content_lower_priority)
+    os.chmod(path_lower_priority, 0o600)
+
+    paths = [path_lower_priority]
+    monkeypatch.setattr(utils, "CONFIG_PATHS", value=paths)
+    # user specified path
+    tool_opts = utils.options_from_config_files(tool_opts, path)
+
+    assert tool_opts.password == output["password"]
+    assert tool_opts.activation_key == output["activation_key"]
+    if "INVALID_HEADER" in content:
+        assert "Unsupported header" in caplog.text
+    if "incorrect_option" in content:
+        assert "Unsupported option" in caplog.text
```

Comment:
Note: I don't see either of these tested in this function.  Perhaps you want to put some of these in the lower priority config sources to test that they won't cause a failure of `utils.options_from_config_files()` in the future.

Reply:
@abadger Not sure what do you mean by this. Is it like if in the config file with lower priority would be typo it shouldn't warn user about the mistake?
Or it's just about adding new scenarios which will test this path? Because I've forgot these asserts there since splitting the test.

Reply:
The latter.  Just adding more scenarios to the tests.